sdk/iot/provisioning/samples/src/paho_iot_provisioning_example.c

```diff
@@ -262,6 +220,119 @@ static int register_device()
   return 0;
 }
 
+static void print_buffer(az_span span)
+{
+  char* buffer = (char*)az_span_ptr(span);
+  for (int i = az_span_size(span); i > 0; i--)
+  {
+    putchar(*buffer++);
+  }
+
+  putchar('\n');
+}
+
+static int get_operation_status()
+{
+  int rc;
+  bool registration_complete = false;
+  char* topicName;
+  int topicLen;
+  MQTTClient_message* message;
+
+  while (!registration_complete)
+  {
+    if ((rc = MQTTClient_receive(mqtt_client, &topicName, &topicLen, &message, 60 * 1000))
+        != MQTTCLIENT_SUCCESS)
+    {
+      printf("Failed to receive message, return code %d\n", rc);
+      return rc;
+    }
+
+    if (topicLen == 0)
+    {
+      // The length of the topic if there are one more NULL characters embedded in topicName,
+      // otherwise topicLen is 0.
+      topicLen = (int)strlen(topicName);
+    }
+
+    printf("Message arrived\n");
+    printf("     topic (%d): %s\n", topicLen, topicName);
+    printf("   message (%d): ", message->payloadlen);
+    print_buffer(az_span_init(message->payload, message->payloadlen));
+
+    az_iot_provisioning_client_register_response response;
+    char query_topic[256];
+
+    if (az_failed(
+            rc = az_iot_provisioning_client_parse_received_topic_and_payload(
+                &provisioning_client,
+                az_span_init((uint8_t*)topicName, topicLen),
+                az_span_init((uint8_t*)message->payload, message->payloadlen),
+                &response)))
+    {
+      printf("Message from unknown topic, return code %d\n", rc);
+      return rc;
+    }
+
+    if (az_span_is_content_equal(response.registration_state, AZ_SPAN_FROM_STR("assigning")))
```

Comment:
>assigning [](start = 80, length = 9)

How about these schema DPS schema related values come from sdk dps header file: like macros defining these DPS schema key values?

Reply:
Good point. I'll need to think about the best way to achieve that: a function adapting to an enum vs defining the strings, etc.